tgui/packages/tgui/interfaces/OreSilo.tsx

```diff
@@ -0,0 +1,199 @@
+            {currentTab === 0 ? (
+              <Section fill scrollable>
+                {machines.map((machine) => (
+                  <MachineDisplay
+                    key={machine.name}
+                    machine={machine}
+                    onPause={(machine) => act('hold', { ref: machine.ref })}
+                    onRemove={(machine) => act('remove', { ref: machine.ref })}
+                  />
+                ))}
+              </Section>
+            ) : null}
+            {currentTab === 1 ? (
+              !!logs && logs.length > 0 ? (
+                <Box pr={1} height="100%" overflowY="scroll">
+                  {logs.map((log, index) => (
+                    <LogEntry key={index} log={log} />
+                  ))}
+                </Box>
+              ) : (
+                <NoticeBox>No log entries currently present!</NoticeBox>
+              )
+            ) : null}
+          </Stack.Item>
+          <Stack.Item>
+            <Section fill>
+              <MaterialAccessBar
+                availableMaterials={data.materials}
+                SHEET_MATERIAL_AMOUNT={SHEET_MATERIAL_AMOUNT}
+                onEjectRequested={(material, amount) =>
+                  act('eject', { ref: material.ref, amount })
+                }
+              />
+            </Section>
+          </Stack.Item>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+interface MachineProps {
+  machine: Machine;
+  onPause: (machine: Machine) => void;
+  onRemove: (machine: Machine) => void;
+}
+
+const MachineDisplay = (props: MachineProps) => {
+  const { machine, onPause, onRemove } = props;
+
+  return (
+    <Box className="FabricatorRecipe">
+      <Box className="FabricatorRecipe__Title">
+        <Box className="FabricatorRecipe__Icon">
+          <Image
+            width={'32px'}
+            height={'32px'}
+            src={`data:image/jpeg;base64,${machine.icon}`}
+          />
+        </Box>
+        <Box className="FabricatorRecipe__Label">{machine.name}</Box>
+      </Box>
+
+      <Tooltip
+        content={
+          machine.onHold
+            ? `Resume ${machine.name} usage.`
+            : `Put ${machine.name} on hold.`
+        }
+      >
+        <Box
+          className={classes([
+            'FabricatorRecipe__Button',
+            'FabricatorRecipe__Button--icon',
+          ])}
+          onClick={(_) => {
+            onPause(machine);
+          }}
+        >
+          <Icon name={machine.onHold ? 'circle-play' : 'circle-pause'} />
+        </Box>
+      </Tooltip>
+      <Tooltip content={`Disconnect ${machine.name}.`}>
+        <Box
+          className={classes([
+            'FabricatorRecipe__Button',
+            'FabricatorRecipe__Button--icon',
+          ])}
+          onClick={(_) => {
+            onRemove(machine);
+          }}
+        >
+          <Icon name="trash-can" />
+        </Box>
+      </Tooltip>
+    </Box>
+  );
+};
+
+interface LogProps {
```

Comment:
I haven't seen tgui code in a while but we don't usually use interface, do we? Should be `type LogProps = {` unless I'm mistaken.

Reply:
Here and everywhere else

Reply:
Sure, but what's the difference between those?